ocs/rook.py

```diff
@@ -0,0 +1,142 @@
+"""
+A module for all rook functionalities and abstractions.
+
+This module has rook related classes, support for functionalities to work with
+rook cluster. This works with assumptions that an OCP cluster is already
+functional and proper configurations are made for interaction.
+"""
+
+import logging
+
+import oc.openshift_ops as ac
+from ocs.pod import Pod
+
+logger = logging.getLogger(__name__)
+
+
+class Rook(object):
+    """
+    High level rook abstraction. This class should handle operator
+    and cluster objects.
+
+    As of now deligated rook environment to corresponding cluster object
+    but in future rookenv should be handled here.Also few operator related ops
+    will be included in this class. Keeping this class lightweight for the first
+    cut.
+
+    Attrs:
+        self._cluster (RookCluster): current cluster object
+    """
+
+    def __init__(self, **config):
+        self._cluster = RookCluster(**config)
+
+    @property
+    def cluster(self):
+        return self._cluster
+
+
+class RookCluster(object):
+    """
+    Handles all cluster related operations from ceph perspective
+
+    This class has depiction of ceph cluster. Contains references to
+    pod objects which represents ceph cluster entities.
+
+    Attributes:
+        _ocs_pods (list) : A list of  ceph cluster related pods
+
+        _cluster_name (str): Name of ceph cluster.
+
+        _namespace (str): openshift Namespace where this cluster lives.
+
+        _api_client (APIClient): api-client used for interacting with openshift.
+
+        _url_prefix (str): url prefix for REST api interaction
+
+    Methods:
+       add_mon(**kw) : add a new mon
+       remove_mon(id): Remove a mon with id id
+       add_osd(**kw):  add a new osd to cluster
+    """
+
+    def __init__(self, **config):
+        # cluster_name is name of cluster in rook of type CephCluster
+        self._cluster_name = config.get('cluster_name', 'rook-ceph')
+        self._namespace = config.get('namespace', 'rook-ceph')
```

Comment:
For cluster namespace we will be using probably openshift-storage as we have it as default in template as you can see:  https://github.com/red-hat-storage/ocs-ci/pull/23/files#diff-788816bc8d86d39b43365dfba5670b65R16 , So would be good to put it to defaults and use it from defaults

Reply:
This one I couldn't figure out where to put it and how to retrieve . Which template file will have this ? Could you please point me to an example.

Reply:
@petr-balogh ,
After further discussion with Vasu, I went ahead with defaults.py approach. Since rook object has to be instantiated somewhere after deployment we thought we will do reading from config there instead of doing it inside RookCluster class.

Reply:
We will have to keep the defaults in defaults.py, I will add the trello on how we can just use one place for defaults and move few things from template to here, redenering the template with defaults is something we have to look into.